MatrixRotator.js

```diff
@@ -11,16 +11,47 @@ const Direction = require("./Direction").Direction;
  *        @returns the rotated matrix
  */
 module.exports = class MatrixRotator {
-  constructor( matrix ) {
+  constructor(matrix) {
     this.matrix = matrix;
   }
 
   //      |-- Must be Direction.CW
   //      v        or Direction.CCW
-  rotate (direction) {
+  rotate(direction) {
     // do work here
 
-    // must be a valid Direction, see Direction.js
+    // make a dummy array to accept the rotated matrix values
+    // cannot make a copy of starting array because changes to either array will affect BOTH arrays
+    let newArray = [];
+    for (let row = 0; row < this.matrix.length; row++) {
```

Comment:
A faster way of copying a matrix would be to use slice for east nested array, and finally slicing the uppermost array.